daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,316 @@
+package firewall
+
+import (
+	"errors"
+	"fmt"
+	"net/netip"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/NordSecurity/nordvpn-linux/daemon/device"
+)
+
+var ErrRuleAlreadyActive = errors.New("this rule is already active")
+
+const (
+	iptables  = "iptables"
+	ip6tables = "ip6tables"
+)
+
+type IptablesExecutor interface {
+	ExecuteCommand(command string) error
+	ExecuteCommandIPv6(command string) error
+}
+
+type Iptables struct {
+	ip6tablesSupported bool
+}
+
+func AreIp6tablesSupported() bool {
+	// #nosec G204 -- input is properly sanitized
+	_, err := exec.Command(ip6tables, "-S").CombinedOutput()
+	return err != nil
+}
+
+func NewIptables() Iptables {
+	return Iptables{
+		ip6tablesSupported: AreIp6tablesSupported(),
+	}
+}
+
+func (i Iptables) ExecuteCommand(command string) error {
+	commandArgs := strings.Split(command, " ")
+
+	// #nosec G204 -- arg values are known before even running the program
+	if _, err := exec.Command(iptables, commandArgs...).CombinedOutput(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i Iptables) ExecuteCommandIPv6(command string) error {
+	if !i.ip6tablesSupported {
+		return errors.New("ip6tables are not supported")
+	}
+
+	commandArgs := strings.Split(command, " ")
+
+	// #nosec G204 -- arg values are known before even running the program
+	if _, err := exec.Command(ip6tables, commandArgs...).CombinedOutput(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type PortRange struct {
+	min int
+	max int
+}
+
+type FirewallManager struct {
+	commandExecutor   IptablesExecutor
+	devices           device.ListFunc // list network interfaces
+	allowlistRules    []string
+	trafficBlockRules []string
+	connmark          uint32
+	enabled           bool
+}
+
+func NewFirewallManager(devices device.ListFunc, commandExecutor IptablesExecutor, connmark uint32, enabled bool) FirewallManager {
+	return FirewallManager{
+		commandExecutor: commandExecutor,
+		devices:         devices,
+		connmark:        connmark,
+		enabled:         enabled,
+	}
+}
+
+func (f *FirewallManager) BlockTraffic() error {
+	if f.trafficBlockRules != nil {
+		return ErrRuleAlreadyActive
+	}
+
+	interfaces, err := f.devices()
+	if err != nil {
+		return fmt.Errorf("listing interfaces: %w", err)
+	}
+
+	// -I INPUT -i <iface> -m comment --comment nordvpn -j DROP
+	// -I OUTPUT -o <iface> -m comment --comment nordvpn -j DROP
+	for _, iface := range interfaces {
+		inputCommand := fmt.Sprintf("INPUT -i %s -m comment --comment nordvpn -j DROP", iface.Name)
+		outputCommand := fmt.Sprintf("OUTPUT -o %s -m comment --comment nordvpn -j DROP", iface.Name)
+		f.trafficBlockRules = append(f.trafficBlockRules, inputCommand)
+		f.trafficBlockRules = append(f.trafficBlockRules, outputCommand)
+
+		if f.enabled {
```

Comment:
when enabled, if one of the commands fails, then `removeBlockTrafficRules` will also fail because  `f.trafficBlockRules` will have rules that are not inserted into the firewall. 
Since the rules are appended at the end of `f.trafficBlockRules` probably the firewall will not remain in a bad state, but `removeBlockTrafficRules` will return error for rule not found.